htdocs/product/stock/replenish.php

```diff
@@ -679,7 +679,12 @@
 		}
 
 		// Force call prod->load_stats_xxx to choose status to count (otherwise it is loaded by load_stock function)
+        $draftordered = 0;
 		if (isset($draftchecked)) {
+            if(!empty($usevirtualstock)){
+                $result = $prod->load_stats_commande_fournisseur(0, '0');
+                $draftordered = $prod->stats_commande_fournisseur['qty'];
+            }
 			$result = $prod->load_stats_commande_fournisseur(0, '0,1,2,3,4');
```

Comment:
Into $prod->load_stats_commande_fournisseur we already have the '0' status here.
So why getting it a second time before ?

Reply:
we need only draft qty to decrease from virtual stock if conf is activated because virtual stock handle every status except this one

Reply:
So, if virtual stock is not returned by load_stats_commande_fournisseur for status '0', even when we provide status '0' as parameter to define the list of status we want, it means the bug is inside the load_stats_commande_fournisseur function, isn't it ?

Reply:
no, there is no bug in  this function, it's just that we don't have the detail of qty by status in this function and i need only qty for draft status.
To explain more : 
$result = $prod->load_stats_commande_fournisseur(0, '0,1,2,3,4');
after that $prod->stats_commande_fournisseur['qty'] will have number for all these statuses.
But virtual stock already contains qty of  1,2,3,4 statuses that's why for virtual stock we just need the qty for status 0

Reply:
Ok, i see.
You need qty for draft status ONLY because the variable $prod->stock_theorique does not includes draft status.
But as the comment say "ordered is (should be) already in $stock in virtual mode". The trouble here is before: When we make "$prod->load_stock('warehouseopen, warehouseinternal');". This return qty without taking into account the parameter $draftchecked.
So you should instead propagate parameter $draftchecked when calling 
$prod->load_stock()
and $prod->load_stock should propagate it when calling ->load_virtual_stock() so this last method can include status 0 into it counting. And result is as expected at the source, instead of making a "compensation call" later.

Note: I also suggest to rename variable $draftchecked into $includedraftpoforvirtual

Reply:
Ok i'll open an other pull request